pkgs/top-level/all-packages.nix

```diff
@@ -20154,490 +20154,107 @@ in
-    };
-
-    virtualboxGuestAdditions = callPackage ../applications/virtualization/virtualbox/guest-additions {
-      virtualbox = pkgs.virtualboxHardened;
+  # This serves as a test for linuxPackages_custom
+  linuxPackages_custom_tinyconfig_kernel = let
+    base = pkgs.linuxPackages.kernel;
+    tinyLinuxPackages = linuxKernel.customPackage {
+      inherit (base) version src;
+      allowImportFromDerivation = false;
+      configfile = pkgs.linuxConfig {
+        makeTarget = "tinyconfig";
+        src = base.src;
+      };
     };
+    in tinyLinuxPackages.kernel;
 
-    wireguard = if lib.versionOlder kernel.version "5.6" then callPackage ../os-specific/linux/wireguard { } else null;
-
-    x86_energy_perf_policy = callPackage ../os-specific/linux/x86_energy_perf_policy { };
-
-    xpadneo = callPackage ../os-specific/linux/xpadneo { };
-
-    zenpower = callPackage ../os-specific/linux/zenpower { };
-
-    inherit (callPackages ../os-specific/linux/zfs {
-        configFile = "kernel";
-        inherit kernel;
-      }) zfsStable zfsUnstable;
-    zfs = zfsStable;
-
-    can-isotp = callPackage ../os-specific/linux/can-isotp { };
-  });
+  # Versioned aliases kept for backwards compatibility ------
+  # Don't add to this list!
+  linuxPackages_rt_5_4 = linuxKernel.packages.linux_rt_5_4;
+  linuxPackages_rt_5_10 = linuxKernel.packages.linux_rt_5_10;
+  linuxPackages_rt_5_11 = linuxKernel.packages.linux_rt_5_11;
+  linuxPackages_rpi1 = linuxKernel.packages.linux_rpi1;
+  linuxPackages_rpi2 = linuxKernel.packages.linux_rpi2;
+  linuxPackages_rpi3 = linuxKernel.packages.linux_rpi3;
+  linuxPackages_rpi4 = linuxKernel.packages.linux_rpi4;
+  linuxPackages_4_4 = linuxKernel.packages.linux_4_4;
+  linuxPackages_4_9 = linuxKernel.packages.linux_4_9;
+  linuxPackages_4_14 = linuxKernel.packages.linux_4_14;
+  linuxPackages_hardkernel_4_14 = linuxKernel.packages.hardkernel_4_14;
+  linuxPackages_4_19 = linuxKernel.packages.linux_4_19;
+  linuxPackages_5_4 = linuxKernel.packages.linux_5_4;
+  linuxPackages_5_10 = linuxKernel.packages.linux_5_10;
+  linuxPackages_5_11 = linuxKernel.packages.linux_5_11;
+  linuxPackages_5_12 = linuxKernel.packages.linux_5_12;
+
+  linux_mptcp_95 = linuxKernel.kernels.linux_mptcp_95;
+  linux_rpi1 = linuxKernel.kernels.linux_rpi1;
+  linux_rpi2 = linuxKernel.kernels.linux_rpi2;
+  linux_rpi3 = linuxKernel.kernels.linux_rpi3;
+  linux_rpi4 = linuxKernel.kernels.linux_rpi4;
+  linux_4_4 = linuxKernel.kernels.linux_4_4;
+  linux_4_9 = linuxKernel.kernels.linux_4_9;
+  linux_4_14 = linuxKernel.kernels.linux_4_14;
+  linux_4_19 = linuxKernel.kernels.linux_4_19;
+  linux_5_4 = linuxKernel.kernels.linux_5_4;
+  linux-rt_5_4 = linuxKernel.kernels.linux_rt_5_4;
+  linux_5_10 = linuxKernel.kernels.linux_5_10;
+  linux-rt_5_10 = linuxKernel.kernels.linux_rt_5_10;
+  linux_5_11 = linuxKernel.kernels.linux_5_11;
+  linux_5_12 = linuxKernel.kernels.linux_5_12;
+  linux-rt_5_11 = linuxKernel.kernels.linux_rt_5_11;
+  # ------
```

Comment:
We need a way to communicate that these names are now legacy. How could we achieve this? I feel like a release note isn't going to reach most users. 

Not something that needs to happen in this PR but needs to be thought about at least.

Reply:
Yes I was hoping that most people use the unversioned packages anyway, and those that do pin versions usually keep an eye on what happens in nixpkgs.
This is a problem in general I think: we lack a reliable channel to communicate changes. Although the release notes are still the closest thing to that we have, I guess.
Maybe we can add traces to these at some point (saying that they are deprecated) while still not breaking old configs.

Reply:
Clearly written eval warnings are definitely the best way of doing this kind of thing IMO.  
Maybe we could make these upcoming permanent renames a special kind of alias which prints a trace, similar to how nixos modues renames work.

I'll have a think about that tonight and open an issue tomorrow but feel free to go ahead and create one without me.

Reply:
I don't think exposing the old `linuxPackages*` package sets is too bad. It would also avoid having users needing to update their configuration.nix

Reply:
not sure what to do with the kernel variants

cc @samueldr